R/utils_insert_db.R

```diff
@@ -152,18 +152,20 @@ upload_package_to_db <- function(name, version, title, description,
 insert_riskmetric_to_db <- function(pkg_name, 
     db_name = golem::get_golem_options('assessment_db_name')){
 
-  if (!isTRUE(getOption("shiny.testmode")))
+  if (!isTRUE(getOption("shiny.testmode"))) {
     riskmetric_assess <-
       riskmetric::pkg_ref(pkg_name,
                           source = "pkg_cran_remote") %>%
       dplyr::as_tibble() %>%
       riskmetric::pkg_assess()
-  else
+  } else {
     riskmetric_assess <-
       test_pkg_assess[[pkg_name]]
+  }
+  metric_weights_df <- dbSelect("SELECT id, name, weight, is_perc FROM metric", db_name)
```

Comment:
I think this is were you were originally using `get_metric_weights()` and why you added `is_perc`. Makes me think even more that `is_perc` should be removed from that function.

Reply:
Actually, is_perc is **only** used in `insert_riskmetric_to_db` to calculate `metric_value`
So we can remove `is_perc` from `get_metric_weights()`

```{r, eval=FALSE}
    metric_value <- case_when(
      "pkg_metric_error" %in% class(riskmetric_assess[[metric$name]][[1]]) ~ "pkg_metric_error",
      metric$name == "dependencies" ~ as.character(length(unlist(as.vector(riskmetric_assess[[metric$name]][[1]][1])))),
      metric$name == "reverse_dependencies" ~ as.character(length(as.vector(riskmetric_assess[[metric$name]][[1]]))),
      metric$is_perc == 1L ~ as.character(round(riskmetric_score[[metric$name]]*100, 2)[[1]]),
      TRUE ~ as.character(riskmetric_assess[[metric$name]][[1]][1:length(riskmetric_assess[[metric$name]])])
    )
```